src/pkg/logger/logger.go

```diff
@@ -0,0 +1,147 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+
+// Package logger implements a log/slog based logger in Zarf.
+package logger
+
+import (
+	"fmt"
+	"io"
+	"log/slog"
+	"os"
+	"strings"
+)
+
+// Level declares each supported log level. These are 1:1 what log/slog supports by default. Info is the default level.
```

Comment:
If we are 1:1 with the slog levels should we just use `slog.Level`? IMO we can always introduce our own levels if needed in the future, but better to stay simple for now

Reply:
logger.Level is part of the config and external API, so if we want to change it in the future then backwards compatibility becomes an issue. It's simpler in the long run to add the type now and always have the option to add more or change its implementation later.

Reply:
Yeah that's a fair point